server/block/carpet.go

```diff
@@ -34,12 +35,12 @@ func (c Carpet) BreakInfo() BreakInfo {
 
 // EncodeItem ...
 func (c Carpet) EncodeItem() (name string, meta int16) {
-	return "minecraft:carpet", int16(c.Colour.Uint8())
+	return fmt.Sprintf("minecraft:%s_carpet", c.Colour.String()), 0
```

Comment:
I wouldn't use fmt.Sprintf in these methods. It is quite slow and they're all strings, so + works too.